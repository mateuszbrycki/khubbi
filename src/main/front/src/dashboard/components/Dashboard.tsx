import React from "react";
import {Event, EventDate, EventForms} from "../../types";
import {List} from "immutable";
import EventsTimeline from "../../events/components/EventsTimeline";
import EventsManagement from "../../events/components/EventsManagement";

export interface DashboardProps {
    readonly events: List<Event>,
    readonly showEventForm: EventForms | null
}

export interface DashboardActionProps {
    readonly loadEvents: () => void,
    readonly openAddEventForm: (type: EventForms) => void,
    readonly closeAddEventForm: () => void
    readonly addNote: (note: string, date: EventDate) => void
    readonly addPhoto: (description: string, photo: File, date: EventDate) => void
}

const Dashboard: React.FC<DashboardProps & DashboardActionProps> = (props) => {
    const {events, showEventForm, loadEvents, openAddEventForm, closeAddEventForm, addNote, addPhoto} = props

    return <>
        <div className="d-flex justify-content-center">
            <EventsManagement showEventForm={showEventForm}
                              openAddEventForm={openAddEventForm}
                              closeAddEventForm={closeAddEventForm}
                              addNote={addNote}
                              addPhoto={addPhoto}
            />
        </div>

        <div className="d-flex justify-content-center">
            <EventsTimeline events={events} loadEvents={loadEvents}/>
        </div>
    </>
}

export default Dashboard