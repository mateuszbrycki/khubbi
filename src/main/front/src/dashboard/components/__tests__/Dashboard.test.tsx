import ShallowRenderer from "react-test-renderer/shallow";
import Dashboard from "../Dashboard";
import {List} from "immutable";

it('renders empty list', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<Dashboard events={List.of()} loadEvents={() => null}/>)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
});

it('renders list of events', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<Dashboard events={List.of({name: "event-1", id: "event-1"}, {name: "event-2", id: "event-2"})}
                               loadEvents={() => null}/>)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
});