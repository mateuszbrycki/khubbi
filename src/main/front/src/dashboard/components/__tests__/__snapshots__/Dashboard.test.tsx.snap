// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of events 1`] = `
<React.Fragment>
  <div
    className="d-flex justify-content-center"
  >
    <EventsManagement
      addNote={[Function]}
      addPhoto={[Function]}
      closeAddEventForm={[Function]}
      openAddEventForm={[Function]}
      showEventForm={null}
    />
  </div>
  <div
    className="d-flex justify-content-center"
  >
    <EventsTimeline
      events={
        Immutable.List [
          {
            "date": EventDate {
              "value": "2024-07-12T20:00Z",
            },
            "id": "event-1",
            "note": "event-1",
          },
          {
            "date": EventDate {
              "value": "2024-07-12T21:00Z",
            },
            "id": "event-2",
            "note": "event-2",
          },
        ]
      }
      loadEvents={[Function]}
    />
  </div>
</React.Fragment>
`;
